src/hazelcore/Builtins.re

```diff
@@ -0,0 +1,181 @@
+/*
+   Built-in functions for Hazel.
+
+   To add a built-in function or constant, write the implementation in the
+   `Impls` module below and add it to the `built-ins` list.
+
+   See the existing ones for reference.
+ */
+
+open Sexplib.Std;
+
+[@deriving sexp]
+type args = list(DHExp.t);
+
+[@deriving sexp]
+type form = (
+  /* eval: */ (args, DHExp.t => EvaluatorResult.t) => EvaluatorResult.t,
+  /* elab: */ DHExp.t,
+);
+
+[@deriving sexp]
+type t = (Var.t, HTyp.t, form);
```

Comment:
let's just put the identifier, type, eval function and elab all in one record with field names instead of broken out like this